### PR TITLE
Fix for windows failure InternalDistributionArchiveSetupPluginFuncTest

### LINKED ISSUE #5324: Fix gradle check on 1.3/1.x
These are the failures in 1.x and 1.3 branch from the latest CI runs on windows -

- [x] InternalDistributionArchiveCheckPluginFuncTest. fails on unexpected notice content
- [x] InternalDistributionArchiveCheckPluginFuncTest. plain class files in distribution tar archives are detected
- [x] InternalDistributionArchiveSetupPluginFuncTest. registered distribution provides archives and directory variant

### PR DESCRIPTION
**Pull request by Poojita-Raj**
Signed-off-by: Poojita Raj <poojiraj@amazon.com>

### Description
Failure on windows for the InternalDistributionArchiveSetupPluginFuncTest is unexpected. Fails with "registered distribution provides archives and directory variant".
This is because -
- tar task execution is skipped on windows.
- plain copy task execution doesn't execute on windows since source files will not be present.

To get the test passing again, we check for expected conditions on windows and non-windows machines.

### Issues Resolved
Resolves #5324 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy [hunk 1]
```diff
@@ -119,53 +118,64 @@ class InternalDistributionArchiveSetupPluginFuncTest extends AbstractGradleFuncT
...
+
+            when:
+            gradleRunner("copyDir", "-Pversion=2.0").build()
+            then: "old content is cleared out"
+            file("producer-tar/build/install/someFile.txt").exists()
+            !file("producer-tar/build/install/snapshot-1.0.txt").exists()
+            file("producer-tar/build/install/snapshot-2.0.txt").exists()
+        } else {
+            when:
+            def result = gradleRunner("copyArchive").build()
```
Comment (reta) [inline buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy:167]: I think we should add a test case for Windows ZIP archive (for Windows platform), to check the archive distribution also works on Windows  (we had issues with that in the past), what do you think @Poojita-Raj?

Comment (Poojita-Raj) [reply buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy:167]: Hey @reta, thanks for the review! Yes, I think we can do that, however do you believe that's a blocker for windows readiness for the 1.3.7 release? Or would it be a nice to have add on?

Comment (reta) [reply buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy:167]: Thanks @Poojita-Raj , I think it is nice to have but certainly not a blocker

**Review by reta — APPROVED** [review]

**Review by mch2 — APPROVED** [review]

File: buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy (hunk 1 above)
Comment (Poojita-Raj) [reply buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy:167]: Thanks @reta, I've opened another issue for your comment to track it, and will merge this in for now.

**Pull request merged**